### DISCUSSION TITLE
Self-supervised pretrained weights

### DISCUSSION THREAD

**Discussion by fleecedragoon** [Project-MONAI/MONAI#7208] [General]
Hello, 

I'm interested in making use of a self-supervised pretrained model on CT imaging. In the [tutorials](https://github.com/Project-MONAI/tutorials/tree/main/self_supervised_pretraining), I found two options: [ViT_UNETR_SSL](https://github.com/Project-MONAI/tutorials/tree/main/self_supervised_pretraining/vit_unetr_ssl#self-supervised-pre-training-tutorial)  or [Pre-trained SwinUNETR Backbone](https://github.com/Project-MONAI/tutorials/tree/main/self_supervised_pretraining/swinunetr_pretrained#pre-trained-swinunetr-backbone-on-50000-3d-volumes).

I have two questions regarding these options:

1) Are the pretrained weights for the ViT_UNETR_SSL available without having to download the COVID dataset and pretraining the modified ViT?

2) In the SwinUNETR tutorial, it seems like it should be possible to download the weights, but the path to the where the weights should be stored does not exist

```python
use_pretrained = True
if use_pretrained:
    resource = (
        "https://github.com/Project-MONAI/MONAI-extra-test-data/releases/download/0.8.1/ssl_pretrained_weights.pth"
    )
    dst = "./ssl_pretrained_weights.pth"
    download_url(resource, dst)
    pretrained_path = os.path.normpath(dst)
```

**Comment by KumoLiu** [comment]
Hi @fleecedragoon, the "ssl_pretrained_weights.pth" is located [here](https://github.com/Project-MONAI/MONAI-extra-test-data/releases). Thanks!

**Reply by fleecedragoon** [reply]
Thanks!

**Reply by SII-EugeneWong** [reply]
Hi @KumoLiu, thanks for sharing the pretrained weight.

I noticed that the `ssl_pretrained_weights.pth` contains keys like:

* `encoder.*`
* `decoder*`
* `out.conv.*`
* `encoder.mask_token`

which seem different from the `SSLHead` model structure in the SwinUNETR self-supervised pretraining code (e.g., the one with `rotation_head`, `contrastive_head`, and `conv`).

Could you clarify **which codebase or training script was used to produce this checkpoint**?

Specifically:

* Was it trained using the **SwinUNETR self-supervised pretraining framework** from the MONAI research contributions repo?
* Or was it trained using a **different encoder-decoder reconstruction framework**?

If possible, could you also share:

* the **training script or repository** used to generate this weight,
* and the **pretraining tasks** (e.g., masked reconstruction, contrastive learning, rotation prediction)?

This would really help understand how the released checkpoint was produced.

Thanks again!

**Comment by fleecedragoon** [comment]
Hi again,

What is the difference between the **model_swinvit.pt** weights, which should correspond to ["the self-supervised pre-trained weights for Swin UNETR backbone"](https://github.com/Project-MONAI/research-contributions/tree/main/SwinUNETR/Pretrain#pre-trained-models) and the **ssl_pretrained_weights.pth** weights, [which also correspond to the pre-trained weights for SwinUNETR](https://github.com/Project-MONAI/tutorials/tree/main/self_supervised_pretraining/swinunetr_pretrained#2-pretrained-swinunetr-weights)?
